vllm_detector_adapter/generative_detectors/granite_guardian.py

```diff
@@ -47,3 +58,121 @@ def preprocess(
+                message=NO_RISK_NAME_MESSAGE,
+                type="BadRequestError",
+                code=HTTPStatus.BAD_REQUEST.value,
+            )
+
+        if len(request.context) > 1:
+            # The detector API for context docs detection supports more than one context text
+            # but currently chat completions will only take one context. Here, we concatenate
+            # multiple contexts together if provided. Models will error if the user request
+            # exceeds the model's context length
+            logger.warning("More than one context provided. Concatenating contexts.")
+        context_text = " ".join(request.context)  # Will not affect single context case
+        content = request.content
+        # The "context" role is not an officially supported OpenAI role, so this is specific
+        # to Granite Guardian. Messages must also be in precise ordering, or model/template
+        # errors may occur.
+        if risk_name in self.RESPONSE_CONTEXT_ANALYSIS_RISKS:
+            # Response analysis
+            messages = [
+                {"role": "context", "content": context_text},
+                {"role": "assistant", "content": content},
+            ]
+        elif risk_name in self.PROMPT_CONTEXT_ANALYSIS_RISKS:
+            # Prompt analysis
+            messages = [
+                {"role": "user", "content": content},
+                {"role": "context", "content": context_text},
+            ]
+        else:
+            # Return error if risk names are not expected ones
+            return ErrorResponse(
+                message="risk_name {} is not compatible with context analysis".format(
+                    risk_name
+                ),
+                type="BadRequestError",
+                code=HTTPStatus.BAD_REQUEST.value,
+            )
+
+        # Try to pass all detector_params through as additional parameters to chat completions
+        # without additional validation or parameter changes, similar to ChatDetectionRequest processing
+        try:
+            return ChatCompletionRequest(
+                messages=messages,
+                model=model_name,
+                **request.detector_params,
+            )
+        except ValidationError as e:
+            return ErrorResponse(
+                message=repr(e.errors()[0]),
+                type="BadRequestError",
+                code=HTTPStatus.BAD_REQUEST.value,
+            )
+
+    async def context_analyze(
+        self,
+        request: ContextAnalysisRequest,
+        raw_request: Optional[Request] = None,
+    ) -> Union[DetectionResponse, ErrorResponse]:
+        """Function used to call chat detection and provide a /context/doc response"""
+        # Fetch model name from super class: OpenAIServing
+        model_name = self.models.base_model_paths[0].name
+
+        # Apply task template if it exists
+        if self.task_template:
+            request = self.apply_task_template(request)
```

Comment:
just realized this while working on this branch. This function is actually not getting used anymore right?